### PR TITLE
fix(kms): imported key ignores environment from arn

### LINKED ISSUE #21464: (kms): add Key.fromKeyId(), and make Key.fromXXX().env.region reflect the region that the key was imported from
### Describe the feature

The ability to import a kms key based on the key id. And, the `env` for all imported keys should be the environment that they came from, not the environment of the current stack. 

### Use Case

When I already have the key id for a key, it is frustrating to have to go and get the ARN for that key id in order to import the key into my Construct. 

```
const importedKey = kms.Key.fromKeyId(this, 'imported-key', 'xxx-xxx');

// should be the source region
importedKey.env.region

const anotherImportedKey = kms.Key.fromKeyId(this, 'imported-key-2', 'aws::us-east-1::.....');

// should be the source region
anotherImportedKey.env.region

```

### Proposed Solution

I am not sure how exactly to implement `fromKeyId`, since you can't parse the ARN directly from the ID. But since `fromKeyArn` is able to get the key id without doing API lookups, I think it should be possible the other way. 

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

all

### Environment details (OS name and version, etc.)

MacOS

### PR DESCRIPTION
**Pull request by kaizencc**
Fixes #21464. KMS keys imported using `fromKeyArn()` currently take the environment of the stack, not the environment from the arn.

This PR follows the precedent set in https://github.com/aws/aws-cdk/pull/19026 and https://github.com/aws/aws-cdk/pull/18255. It is essentially the same code change and tests. Ideally, we would have a mechanism for testing all `fromXxxArn` APIs to ensure they have the correct behavior. There are still many places where it does not. However, given the significant overhead of creating such a mechanism, I'm creating this one-off PR to unblock users in KMS.

----

### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Given that this issue wasn't picked up by an integration test, I think one should be added to verify the fix.

**Comment by kaizencc** [comment]
I don't believe this needs an integ test. It's also impossible to integ test since it would require a pre-existing kms Key to be referenced with `fromKeyArn`. Instead, the unit tests should be sufficient to test the behavior and has precedent from the similar PRs in [lambda](https://github.com/aws/aws-cdk/pull/18255) and [stepfunctions](https://github.com/aws/aws-cdk/pull/19026).

File: packages/@aws-cdk/aws-kms/test/key.test.ts [hunk 1]
```diff
@@ -1240,3 +1240,34 @@ describe('key specs and key usages', () => {
...
+  let stack: cdk.Stack;
+
+  beforeEach(() => {
+    const app = new cdk.App();
+    stack = new cdk.Stack(app, 'Base', {
+      env: { account: '111111111111', region: 'stack-region' },
+    });
+  });
+
+  describe('for a state machine in a different account and region', () => {
```
Comment (madeline-k) [inline packages/@aws-cdk/aws-kms/test/key.test.ts:1254]: Update description here, this is not longer in step functions ;)

**Review by madeline-k — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-kms/test/key.test.ts (hunk 1 above)
Comment (kaizencc) [reply packages/@aws-cdk/aws-kms/test/key.test.ts]: 😂

**Review by madeline-k — APPROVED** [review]

**Review by TheRealAmazonKendra — APPROVED** [review]

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio requeue

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio requeue

**Pull request merged**